File: Backend/Models/Log.cs
using Microsoft.EntityFrameworkCore;

namespace Backend.Models;

[Owned]
public class Log
{
    public required Rating Rating { get; init; }
    public required State State { get; init; }
    public required DateTime Due { get; init; }
    public required float Stability { get; init; }
    public required float Difficulty { get; init; }
    public required int ElapsedDays { get; init; }
    public required int LastElapsedDays { get; init; }
    public required int ScheduledDays { get; init; }
    public required DateTime Review { get; init; }
}

File: Frontend/src/lib/models/Log.ts
import { Rating, State } from 'ts-fsrs';

export interface Log {
	userID: string;
	learningSessionId: string;
	reviewId: string;
	log: {
		rating: Rating;
		state: State;
		due: Date;
		stability: number;
		difficulty: number;
		elapsed_days: number;
		last_elapsed_days: number;
		scheduled_days: number;
		review: Date;
	};
}
